file/builder.go

```diff
@@ -381,6 +403,37 @@ func (b *stateBuilder) ingestACLGroups(creds []kong.ACLGroup) error {
 	return nil
 }
 
+func (b *stateBuilder) ingestMTLSAuths(creds []kong.MTLSAuth) error {
+	for _, cred := range creds {
+		cred := cred
+		if utils.Empty(cred.ID) {
+			// normally, we'd want to look up existing resources in this case
+			// however, this is impossible here: mtls-auth simply has no unique fields other than ID,
+			// so we just give up and complain about it
+			var consumerFriendlyName *string
+			if !utils.Empty(cred.Consumer.Username) {
+				consumerFriendlyName = cred.Consumer.Username
+			} else if !utils.Empty(cred.Consumer.CustomID) {
+				consumerFriendlyName = cred.Consumer.CustomID
+			} else {
+				// this shouldn't happen, but apparently can. failsafe in case:
+				consumerFriendlyName = cred.Consumer.ID
+			}
+			return errors.Errorf("mtls-auth for Consumer '%s' with SubjectName '%s' lacks ID",
+				*consumerFriendlyName, *cred.SubjectName)
+		}
+		// we only have these from eariler to print errors if needed
+		// they shouldn't go into the actual entity, so strip them out
+		cred.Consumer.CustomID = nil
+		cred.Consumer.Username = nil
+		if b.kongVersion.GTE(kong140Version) {
+			utils.MustMergeTags(&cred, b.selectTags)
```

Comment:
Similar to @mmorel-35's comment on the table name, but more annoying. You're probably right (I didn't check), but why are these credentials not tagged whereas others are? I can't think of an obvious reason; it seems like a possible omission. That seems like something we'd want Fast Track to address, since it brings up issues with managing them via both decK and the controller.

Reply:
Alright, have requested that in FTI-1706, but we won't have it for a while. I think I can just comment out this code with a note to uncomment it once we do get tags, but the codegen in schema.go still says mtls-auths have tags. @hbagdi do you know where it's getting that from? I'm not seeing it after a brief review.

Reply:
>it's getting that from?

https://github.com/hbagdi/go-kong/blob/e6e45561dff1941aa936214cbc50dd156af0f0cf/kong/credentials.go#L93

Reply:
Please remove this if block altogether.
Using select-tag functionality with mtls auth results in:

```
creating mtls-auth foo for consumer f1b4d81f-e440-47e2-bc55-16a2494c3a2d
Error: 1 errors occurred:
        while processing event: {Create} failed: 400 Bad Request {"message":"schema violation (tags: unknown field)","name":"schema violation","fields":{"tags":"unknown field"},"code":2}
```